appinventor/appengine/war/index.jsp

```diff
@@ -1,3 +1,20 @@
+<%@page import="com.google.appinventor.server.Server,com.google.appinventor.common.version.AppInventorFeatures" %>
+<%
+   if (request.getScheme().equals("http") && Server.isProductionServer()
+       && AppInventorFeatures.enableHttpRedirect()) {
+        String qs = request.getQueryString();
+        String host = request.getServerName();
+        if (qs != null) {
+           String redirect = "https://" + host + "/?" + qs;
+           response.sendRedirect(redirect);
+        } else {
+           String redirect = "https://" + host;
+           response.sendRedirect(redirect);
+        }
+     return;
+   }
+   response.setHeader("Strict-Transport-Security", "max-age=3600");
```

Comment:
I need to double check this, but wouldn't we want the header set inside of the if statement before we send the redirect, rather than doing it here where it might be in the dev server (and therefore HTTPS isn't set up).

Reply:
After reviewing the HSTS page on MDN it seems like this should work in both the dev and prod services correctly since the header is ignored until HTTPS is used at least once with a valid certificate chain.